Neos.Fusion/Classes/Core/RuntimeFactory.php

```diff
@@ -39,24 +45,38 @@ public function create(array $fusionConfiguration, ControllerContext $controller
         if ($controllerContext === null) {
             $controllerContext = self::createControllerContextFromEnvironment();
         }
-        return new Runtime(
+        $defaultContextVariables = EelUtility::getDefaultContextVariables(
+            $this->defaultContextConfiguration ?? []
+        );
+        $runtime = new Runtime(
             FusionConfiguration::fromArray($fusionConfiguration),
-            $controllerContext
+            FusionDefaultContextVariables::fromArray(
+                ['request' => $controllerContext->getRequest(), ...$defaultContextVariables]
+            )
         );
+        $runtime->setControllerContext($controllerContext);
+        return $runtime;
     }
 
-    public function createFromConfiguration(FusionConfiguration $fusionConfiguration, ControllerContext $controllerContext): Runtime
-    {
-        return new Runtime($fusionConfiguration, $controllerContext);
+    public function createFromConfiguration(
+        FusionConfiguration $fusionConfiguration,
+        FusionDefaultContextVariables $additionalDefaultContextVariables
+    ): Runtime {
```

Comment:
todo: discuss signature with @bwaidelich to be forward compatible to new action uri building stuff https://github.com/neos/flow-development-collection/pull/3002#issuecomment-1503762615

Reply:
we could in the future possibly do this:

```
FusionDefaultContextVariables::fromArray(["actionUriBuilder" => $builder, "request" => $request])
```